### PR TITLE
lfu cache implementation using priority queue

### PR DESCRIPTION
**Pull request by the-pinbo**
I've added the implemented LFU cache using priority queue, and also written a test file which covers most of the edge cases and some additional functions to the linked list package 
#531

### CODE REVIEW

File: cache/lfu/lfu.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		return
+	}
+
+	item.counter++
+	item.value = value
+	item.timeStamp = c.timeStamp
+	c.pq.Update(item)
+}
+
+type PriorityQueue []*Item
```
Comment (tjgurwara99) [inline cache/lfu/lfu.go:91]: This doesn't belong in this package. This is a general structure that should be in the structure package not here.

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: cache/lfu/lfu.go (hunk 1 above)
Comment (the-pinbo) [reply cache/lfu/lfu.go:91]: Go repo does not have an implementation of priority queue.I'll first make a PR for that in structure package.
